### PR TITLE
feat: add topology spread constraints to control how Pods are spread

### LINKED ISSUE #261: Allow configuring topologySpreadConstraints in Helm chart
Documentation: [Pod Topology Spread Constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/)

topologySpreadConstraints, unlike pod anti-affinity, allow even distribution among of pods among a topology domain, e.g. availability zones.

Configuring them in the chart is not possible at this time.

### PR DESCRIPTION
**Pull request by sshann07**
## what

- It's not possible to spread pod over different defined zones with topologySpreadConstraints at the moment.

## why

- Add the possibility to use topologySpreadConstraints in the helm chart.

## tests

- [x] I have tested my changes by creating an image and configure that image for my atlantis setup in different repos

## references

- closes #261
 - https://kubernetes.io/docs/concepts/scheduling-eviction/topology-spread-constraints/

### CODE REVIEW

File: charts/atlantis/Chart.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: v0.22.3
 description: A Helm chart for Atlantis https://www.runatlantis.io
 name: atlantis
-version: 4.10.3
+version: 4.10.4
```
Comment (nitrocode) [inline charts/atlantis/Chart.yaml:5]:
```suggestion
version: 4.11.0
```

Comment (sshann07) [reply charts/atlantis/Chart.yaml]: version has been upgraded as per the suggestion.

**Comment by nitrocode** [comment]
cc: @runatlantis/maintainers one more reviewer please

**Review by nitrocode — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @sshann07!

**Pull request merged**